aiida/tools/groups/paths.py

```diff
@@ -268,8 +247,7 @@ def walk(self, return_virtual=True):
                 if return_virtual or not sub_child.is_virtual:
                     yield sub_child
 
-    def walk_nodes(self, filters=None, node_class=None, query_batch=None):
-        # type: () -> Iterable[WalkNodeResult]
+    def walk_nodes(self, filters=None, node_class=None, query_batch=None) -> Iterable[WalkNodeResult]:
```

Comment:
```suggestion
    def walk_nodes(self, filters: Optional[dict]=None, node_class: Optional[orm.Node]=None, query_batch: Optional[int]=None) -> Iterable[WalkNodeResult]:
```

so is this not allowed in 3.5?

Reply:
That.. should all be fine I think. What's not allowed is hints directly on a _variable_ (not function argument), and classes which are not yet defined must be given as string.

Reply:
Of course, you could make it an `Optional[Dict[key_type, value_type]]` if you wanna be cute.

Reply:
Well what would be super cute would be using `TypeDicts` to create a `filters` interface, which would in turn give much improved autocompletion for query writing. But that's for next year!

Reply:
(btw I didn't change `Iterable` to `Iterator` here)